### PR TITLE
[23.2] Support configuring job metrics inline, update documentation

### PR DESCRIPTION
**Pull request by natefoo**
YAML config has long been possible but not well documented.

Documents cgroupsv2 but that's implemented in a separate PR #17169

## How to test the changes?
- [x] This is a refactoring of components with existing test coverage.

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

### CODE REVIEW

File: lib/galaxy/job_metrics/__init__.py [hunk 1]
```diff
@@ -68,10 +69,17 @@ class RawMetric(NamedTuple):
...
         """Load :class:`JobInstrumenter` objects from specified configuration file."""
         self.plugin_classes = self.__plugins_dict()
-        self.default_job_instrumenter = JobInstrumenter.from_file(self.plugin_classes, conf_file, **kwargs)
+        if conf_file and os.path.exists(conf_file):
+            self.default_job_instrumenter = JobInstrumenter.from_file(self.plugin_classes, conf_file, **kwargs)
+        elif conf_dict or conf_dict is None:
+            if conf_dict is None:
+                conf_dict = DEFAULT_CONFIG
+            self.default_job_instrumenter = JobInstrumenter.from_dict(self.plugin_classes, conf_dict, **kwargs)
+        else:
```
Comment (jdavcs) [inline lib/galaxy/job_metrics/__init__.py:81]: Is this reachable?  Doesn't the previous clause cover all possibilities? (conf_file is either None or not None, if we don't count the possibility of `false`, the empty string and zero?)

Comment (natefoo) [reply lib/galaxy/job_metrics/__init__.py:81]: It covers any non-`None` falsey values so you can set `job_metrics: false` or `job_metrics: []` and disable the default behavior of getting the `core` plugin if nothing is configured.

Comment (jdavcs) [reply lib/galaxy/job_metrics/__init__.py:81]: Got it! Would you be terribly opposed to adding an inline comment that clarifies this? (for folks like myself :)

**Review by jdavcs — APPROVED** [review]

File: lib/galaxy/job_metrics/__init__.py (hunk 1 above)
Comment (natefoo) [reply lib/galaxy/job_metrics/__init__.py:81]: Done, plus I added it to the docs for good measure.

Comment (jdavcs) [reply lib/galaxy/job_metrics/__init__.py:81]: Awesome, thanks!

**Pull request merged**